spec.md

```diff
@@ -1379,6 +1385,109 @@ For success responses, the following fields are defined:
 }
 ```
 
+## Fetching a Service Instance
+
+If `"instances_retrievable" :true` is declared for a service in the
+[Catalog](#catalog-management) endpoint, Service Brokers MUST support this
+endpoint for all plans of the service.
+
+### Request
+
+##### Route
+`GET /v2/service_instances/:instance_id`
+
+`:instance_id` is the identifier of a previously provisioned instance.
+
+##### cURL
+```
+$ curl 'http://username:password@broker-url/v2/service_instances/:instance_id' -X GET -H "X-Broker-API-Version: 2.13"
+```
+
+### Response
+
+| Status Code | Description |
+| --- | --- |
+| 200 OK | The expected response body is below. |
+| 404 Not Found | MUST be returned if the Service Instance does not exist or if a provisioning operation is still in progress. |
+
+Responses with any other status code will be interpreted as a failure.
+
+##### Body
+
+For success responses, the following fields are defined:
+
+| Response field | Type | Description |
+| --- | --- | --- |
+| service_id | string | The ID of the service from the catalog that is associated with the Service Instance. |
+| plan_id | string | The ID of the plan from the catalog that is associated with the Service Instance. |
+| dashboard_url | string | The URL of a web-based management user interface for the Service Instance; we refer to this as a service dashboard. The URL MUST contain enough information for the dashboard to identify the resource being accessed (`9189kdfsk0vfnku` in the example below). Note: a Service Broker that wishes to return `dashboard_url` for a Service Instance MUST return it with the initial response to the provision request, even if the service is provisioned asynchronously. |
+| parameters | object | Configuration parameters for the Service Instance. |
+
+```
+{
+  "dashboard_url": "http://example-dashboard.example.com/9189kdfsk0vfnku",
+  "parameters": {
+    "billing-account": "abcde12345"
+  }
+}
+```
+
+## Fetching a Service Binding
+
+If `"bindings_retrievable" :true` is declared for a service in the
+[Catalog](#catalog-management) endpoint, Service Brokers MUST support this
+endpoint for all plans of the service.
+
+### Request
+
+##### Route
+`GET /v2/service_instances/:instance_id/service_bindings/:binding_id`
+
+The `:instance_id` is the ID of a previously-provisioned Service Instance. The
+`:binding_id` is the ID of a previously provisioned binding for that instance.
+
+##### cURL
+```
+$ curl 'http://username:password@broker-url/v2/service_instances/:instance_id/service_bindings/:binding_id' -X GET -H "X-Broker-API-Version: 2.13"
+```
+
+### Response
+
+| Status Code | Description |
+| --- | --- |
+| 200 OK | The expected response body is below. |
+| 404 Not Found | MUST be returned if the Service Binding does not exist or if a binding operation is still in progress. |
+
+Responses with any other status code will be interpreted as a failure.
+
+##### Body
+
+For success responses, the following fields are defined:
```

Comment:
Does "the following fields are defined" imply that all of the fields are optional or mandatory? I think it'd be clearer if we were more explicit. (Same with the instance parameter fields above). In the descriptions below, it seems to say "if this feature is not defined in the catalog, then including this field is invalid", but if the feature is defined in the catalog, is it a requirement that this field be present? Is it allowed to return multiple of these fields?

Reply:
The optionality of a field is defied by whether it is marked with an asterisk or not; * == required.  By using the word "defined" we're saying that the spec has some well-known fields w/o implying that extension fields can't also be defined.  Which specific text are you referring to in the 2nd part of your question?